TM1py/Services/CubeService.py

```diff
@@ -224,6 +250,21 @@ def search_for_dimension_substring(self, substring: str, skip_control_cubes: boo
         response = self._rest.GET(url, **kwargs)
         cube_dict = {entry['Name']: [dim['Name'] for dim in entry['Dimensions']] for entry in response.json()['value']}
         return cube_dict
+    
+    def search_for_rule_substring(self, substring: str, skip_control_cubes: bool = False, **kwargs) -> List[Cube]:
+        """ get all cubes from TM1 Server as TM1py.Cube instances where rules for given cube contain specified substring
+
+        :param substring: string to search for in rules
+        :param skip_control_cubes: bool, True will exclude control cubes from result
+        :return: List of TM1py.Cube instances
+        """
+        url = format_url(
+            "/api/v1/{}?$filter=Rules ne null and contains(toupper(Rules),toupper('{}'))&$expand=Dimensions($select=Name)",
```

Comment:
I wonder if we would like the `contains` check to be space insensitive as well. 
Any thoughts?

Reply:
I think that makes sense and will update the odata filter accordingly. Makes me wonder then if we should update the search_string_in_code function in ProcessService to also be space insensitive?

Reply:
Thanks.

Yeah. I also think it would also be nice to keep it consistent.
Let's do this in a separate PR before the next release.